x/feedist/handler.go

```diff
@@ -0,0 +1,141 @@
+package feedist
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/x"
+)
+
+const (
+	distributeCost    = 0
+	updateRevenueCost = 0
+)
+
+// RegisterQuery registers feedlist buckets for querying.
+func RegisterQuery(qr weave.QueryRouter) {
+	NewRevenueBucket().Register("revenues", qr)
+}
+
+// RegisterRouter registers handlers for feedlist message processing.
+func RegisterRouter(r weave.Registry, auth x.Authenticator) {
+	bucket := NewRevenueBucket()
+	r.Handle(pathDistributeMsg, &distributeHandler{auth: auth, bucket: bucket})
+	r.Handle(pathUpdateRevenueMsg, &updateRevenueHandler{auth: auth, bucket: bucket})
+}
+
+type distributeHandler struct {
+	auth   x.Authenticator
+	bucket *RevenueBucket
+}
+
+func (h *distributeHandler) Check(ctx weave.Context, db weave.KVStore, tx weave.Tx) (weave.CheckResult, error) {
+	var res weave.CheckResult
+	if _, err := h.validate(ctx, db, tx); err != nil {
+		return res, err
+	}
+	res.GasAllocated += distributeCost
+	return res, nil
+}
+
+func (h *distributeHandler) Deliver(ctx weave.Context, db weave.KVStore, tx weave.Tx) (weave.DeliverResult, error) {
+	var res weave.DeliverResult
+	msg, err := h.validate(ctx, db, tx)
+	if err != nil {
+		return res, err
+	}
+
+	rev, err := h.bucket.GetRevenue(db, msg.RevenueID)
+	if err != nil {
+		return res, err
+	}
+
+	if err := distribute(ctx, db, rev); err != nil {
+		return res, errors.Wrap(err, "cannot distribute")
+	}
+	return res, nil
+}
+
+func (h *distributeHandler) validate(ctx weave.Context, db weave.KVStore, tx weave.Tx) (*DistributeMsg, error) {
+	rmsg, err := tx.GetMsg()
```

Comment:
I would love to generalize a function to do this boilerplate (get, cast, validate, return), but no idea how to without generics... If you have an idea to make this less boilerplate-tedious, that would be nice for another PR